Adapt to using the circuit arithmetics library

Cryoris opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

Qiskit/qiskit-terra#4062 moved the arithmetic operations we had in `qiskit/aqua/circuits` to circuit arithmetics library and converted them to circuit objects. This PR changes the circuit factory objects in Aqua to use the library objects and deprecates them.

Probably we should leave the tests we had in for now to make sure everything still works and then remove then in a follow-up PR. Or can we already remove them in this one?

woodsp-ibm commented:
We should leave the tests to ensures the components still continue to work even though they are deprecated. In the tests make sure they filter out the deprecation warning i.e. `warnings.filterwarnings("ignore", category=DeprecationWarning)`

Can you also change any components that continue to use these such as european_call_delta and others. (Search for any imports of qiskit.aqua.circuits in the main code and/or look at the build output here as recently we added logic to collate deprecated warnings emitted by the build to ensure Aqua does not use deprecated function - outside of and tests that need to test it, hence the filter).

Also can you create a changelog entry noting that these have become part of new circuit library in Terra etc.

Cryoris commented:
Ok, the tests are still there 👍 I added the deprecation warnings filter to the tests and replaced the places where the previous circuit factory types were used by the circuit library objects (European Call Delta etc.). 
Is the note in the changelog sufficient?
